runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java

```diff
@@ -53,149 +56,167 @@
+        ItemID.STOLEN_GARNET_RING, ItemID.STOLEN_CIRCLET, ItemID.STOLEN_FAMILY_HEIRLOOM, ItemID.STOLEN_JEWELRY_BOX);
+
+    @Inject
+    private Client client;
+
+    @Inject
+    private ClientThread clientThread;
+
+    @Inject
+    private OverlayManager overlayManager;
+
+    @Inject
+    private StealingArtefactsOverlay overlay;
+
+    @Getter(AccessLevel.PACKAGE)
+    private StealingArtefactState stealingArtefactState;
+
+    @Getter(AccessLevel.PACKAGE)
+    private boolean inPortPiscariliusRegion;
+
+    private NPC captainKhaled;
+
+    @Override
+    protected void startUp()
+    {
+
+        overlayManager.add(overlay);
+
+        if (client.getGameState() == GameState.LOGGED_IN)
+        {
+            clientThread.invoke(() -> stealingArtefactState = StealingArtefactState.getState(client.getVar(Varbits.STEALING_ARTEFACT_STATE)));
+        }
+
+    }
+
+    @Override
+    protected void shutDown()
+    {
+        client.clearHintArrow();
+        stealingArtefactState = null;
+        captainKhaled = null;
+        overlayManager.remove(overlay);
+    }
+
+    private boolean hasArtefact()
+    {
+        ItemContainer inventory = client.getItemContainer(InventoryID.INVENTORY);
+
+        if (inventory != null)
+        {
+            final Stream<Item> items = Arrays.stream(inventory.getItems());
+            return items.anyMatch(item -> ARTEFACT_ITEM_IDS.contains(item.getId()));
+        }
+        return false;
+    }
+
+    @Subscribe
+    public void onVarbitChanged(VarbitChanged event)
+    {
+        StealingArtefactState state = StealingArtefactState.getState(client.getVar(Varbits.STEALING_ARTEFACT_STATE));
+
+        if (state != StealingArtefactState.DELIVER_ARTEFACT)
+        {
+            stealingArtefactState = state;
+            if (state == StealingArtefactState.NO_TASK || state == StealingArtefactState.FAILURE)
+            {
+                client.clearHintArrow();
+            }
+            else
+            {
+                client.setHintArrow(stealingArtefactState.getWorldPoint());
+            }
+        }
+        else if (captainKhaled != null)
+        {
+            client.setHintArrow(captainKhaled);
+        }
+        else
+        {
+            client.clearHintArrow();
+        }
+    }
+
+    @Subscribe
+    public void onItemContainerChanged(final ItemContainerChanged event)
+    {
+        if (event.getItemContainer() == client.getItemContainer(InventoryID.INVENTORY))
+        {
+            if (hasArtefact())
+            {
+                stealingArtefactState = StealingArtefactState.DELIVER_ARTEFACT;
+            }
+            else if (stealingArtefactState == StealingArtefactState.DELIVER_ARTEFACT)
+            {
+                stealingArtefactState = StealingArtefactState.FAILURE;
+            }
+        }
+    }
+
+    @Subscribe
+	public void onGameStateChanged(GameStateChanged event)
 	{
-		client.clearHintArrow();
-		stealingArtefactState = null;
 		captainKhaled = null;
```

Comment:
I guess don't clear it if the state is LOGGED_IN. Otherwise, this should be fine.
```suggestion
		if (event.getGameState() != GameState.LOGGED_IN)
		{
			captainKhaled = null;
		}
```